src/mbgl/gl/shader.cpp

```diff
@@ -21,20 +21,24 @@ Shader::Shader(const char* name_,
                const char* vertexSource,
                const char* fragmentSource,
                Context& context,
-               Defines defines)
+               gl::ShaderParameters& shaderParameters)
     : name(name_),
       program(context.createProgram()),
       vertexShader(context.createVertexShader()),
       fragmentShader(context.createFragmentShader()) {
     util::stopwatch stopwatch("shader compilation", Event::Shader);
 
-    if (!compileShader(vertexShader, vertexSource)) {
+    std::string vertex(vertexSource);
+    vertex.replace(vertex.find_first_of('\n'), 1,"\n#define DEVICE_PIXEL_RATIO " + std::to_string(shaderParameters.pixelRatio) + "\n");
+    if (!compileShader(vertexShader, vertex.c_str())) {
         Log::Error(Event::Shader, "Vertex shader %s failed to compile: %s", name, vertexSource);
         throw util::ShaderException(std::string { "Vertex shader " } + name + " failed to compile");
     }
 
+
     std::string fragment(fragmentSource);
-    if (defines & Defines::Overdraw) {
+    fragment.replace(fragment.find_first_of('\n'), 1,"\n#define DEVICE_PIXEL_RATIO "+ std::to_string(shaderParameters.pixelRatio) + "\n");
+    if ((strcmp(name, "collision_box") != 0) & shaderParameters.overdraw) {
```

Comment:
Because I couldn't figure out how to modify things such that the `collision_box_shader` is always initialized with `overdraw = false` 😳